src/main/theme/Resources/modules/color/components/color-chart.jsx

```diff
@@ -6,23 +6,34 @@ import {implementPropTypes} from '#/main/app/prop-types'
 import {trans} from '#/main/app/intl/translation'
 import {CallbackButton} from '#/main/app/buttons/callback/components/button'
 
-import {ColorChartLibrary} from '#/main/theme/color/components/view/library'
-import {ColorChartSelector} from '#/main/theme/color/components/view/selector'
+import {ColorChartLibrary} from '#/main/theme/color/components/library'
+import {ColorChartSelector} from '#/main/theme/color/components/selector'
 import {ColorChart as ColorChartTypes} from '#/main/theme/color/prop-types'
 
 class ColorChart extends Component {
   constructor(props) {
     super(props)
 
     this.state = {
-      view: 'library'
+      view: 'library',
+      viewLocked: false,
+      colorCharts: undefined
```

Comment:
- Pas besoin de stocker tes color charts dans le state react. Tu peux les récupérer directement depuis le store redux. Idem dans ColorChartLibrary
- Pareil pour `viewLocked`, à récupérer direct depuis les props (tu peux faire le `this.props.viewLocked || this.props.colorChart.data.length === 0` qui se trouve plus loin directement dans le `render`, ce n'est pas une opération couteuse, on n'a pas besoin de stocker le résultat)